app/Http/Controllers/PhotoEditorController.php

```diff
@@ -4,35 +4,38 @@
 
 namespace App\Http\Controllers;
 
-use App\Actions\Photo\Prepare;
 use App\Actions\Photo\Rotate;
 use App\Http\Requests\PhotoRequests\PhotoIDRequest;
 use App\Models\Configs;
 use App\Models\Photo;
+use Illuminate\Validation\Rule;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
 
 class PhotoEditorController extends Controller
 {
 	/**
 	 * Given a photoID and a direction (+1: 90° clockwise, -1: 90° counterclockwise) rotate an image.
 	 *
 	 * @param PhotoIDRequest $request
+	 * @param Rotate         $rotate
 	 *
-	 * @return array|string
+	 * @return Photo
 	 */
-	public function rotate(PhotoIDRequest $request, Rotate $rotate, Prepare $prepare)
+	public function rotate(PhotoIDRequest $request, Rotate $rotate): Photo
 	{
 		if (!Configs::get_value('editor_enabled', '0')) {
-			return 'false';
+			throw new UnprocessableEntityHttpException('support for rotation disabled by configuration');
```

Comment:
Create a new Exception inheriting from `UnprocessableEntityHttpException`? 
:thinking:

Reply:
Same issue as in [review 703533052](https://github.com/LycheeOrg/Lychee/pull/1055#pullrequestreview-703533052). I agree that the complete error handling stuff needs to be clean-up, but for me this would be an extra PR which also cleans up the middleware. We should keep the discussion in [review 703533052](https://github.com/LycheeOrg/Lychee/pull/1055#pullrequestreview-703533052) and then come up with a consistent decision for all these issues.